pkg/engine/engine.go

```diff
@@ -242,7 +266,55 @@ func (e *Engine) DetectorAvgTime() map[string][]time.Duration {
 	return avgTime
 }
 
+func (e *Engine) detectChunks(ctx context.Context) {
+	for data := range e.detectableChunksChan {
+		results, err := func() ([]detectors.Result, error) {
+			ctx, cancel := context.WithTimeout(ctx, time.Second*10)
+			defer cancel()
+			defer common.Recover(ctx)
+			return data.detector.FromData(ctx, data.chunk.Verify, data.chunk.Data)
+		}()
+		if err != nil {
+			ctx.Logger().Error(err, "could not scan chunk")
+			continue
+		}
+
+		if e.filterUnverified {
+			results = detectors.CleanResults(results)
+		}
+		for _, result := range results {
+			resultChunk := data.chunk
+			if SupportsLineNumbers(data.chunk.SourceType) {
+				copyChunk := data.chunk
+				copyMetaDataClone := proto.Clone(data.chunk.SourceMetadata)
+				if copyMetaData, ok := copyMetaDataClone.(*source_metadatapb.MetaData); ok {
+					copyChunk.SourceMetadata = copyMetaData
+				}
+				fragStart, mdLine := FragmentFirstLine(&copyChunk)
+				SetResultLineNumber(&copyChunk, &result, fragStart, mdLine)
+				resultChunk = copyChunk
+			}
+			result.DecoderType = data.decoder
+			e.results <- detectors.CopyMetadata(&resultChunk, result)
+
+		}
+		if len(results) > 0 {
+			detectorName := results[0].DetectorType.String()
+			avgTimeI, ok := e.detectorAvgTime.Load(detectorName)
+			var avgTime []time.Duration
+			if ok {
+				avgTime, ok = avgTimeI.([]time.Duration)
+				if !ok {
+					continue
+				}
+			}
+			e.detectorAvgTime.Store(detectorName, avgTime)
+		}
+		data.wgDoneFn()
```

Comment:
The `continue` codepath means `data.wgDoneFn()` doesn't get called. I think the `if` statements could be refactored to:

```go
avgTimeI, ok := e.detectorAvgTime.Load(detectorName)
if ok {
    if avgTime, ok := avgTimeI.([]time.Duration); ok {
        e.detectorAvgTime.Store(detectorName, avgTime)
    }
}
```